### PR TITLE
Backport: Navigation Post Preloading in Site Editor

### PR DESCRIPTION
**Pull request by getdave**
Adds a `navigation.php` file to hold all hooks/utilites relating to Navigation in the Block Editor.

Backports over the preloading hook for Navigation Posts.

Trac ticket: https://core.trac.wordpress.org/ticket/58556

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/navigation.php
Comment (audrasjb) [inline src/wp-includes/navigation.php]: There is an indentation issue on the params :)

**Review by audrasjb — CHANGES_REQUESTED** [review]
I left a small feedback concerning a small docblock issue :)

File: src/wp-includes/navigation.php
Comment (getdave) [reply src/wp-includes/navigation.php]: Thanks for the review. 

I'm afraid I'm not following this feedback. The params seems to be aligned and [the PHPCS CI tests on this PR](https://github.com/WordPress/wordpress-develop/actions/runs/5290055686) pass. 

What needs to change here?

**Comment by spacedmonkey** [comment]
@getdave This new file needs to be loaded in wp-settings.php

**Comment by getdave** [comment]
Understood. My first time doing backports. Let me get this sorted.

**Comment by spacedmonkey** [comment]
> Understood. My first time doing backports. Let me get this sorted.

@getdave See my feedback on trac ticket before continuing.

File: src/wp-includes/navigation.php
Comment (audrasjb) [reply src/wp-includes/navigation.php]: There are 3 unwanted spaces after `$preload_paths` and `$context` :)

**Comment by getdave** [comment]
Preload paths for Navigation have now be inlined directly in the Site Editor file as per recommendations.

**Review by spacedmonkey — APPROVED** [review]

File: src/wp-admin/site-editor.php
Comment (spacedmonkey) [inline src/wp-admin/site-editor.php:103]: The javascript should be refactored so this rest api call is not made twice.

Comment (getdave) [reply src/wp-admin/site-editor.php:103]: Agreed. We're tracking it in https://github.com/WordPress/gutenberg/issues/51803

**Review by audrasjb — APPROVED** [review]

**Comment by getdave** [comment]
Not sure what's going on with the CI tests here. 

I believe I now wait for a Core Committer to merge this?

**Comment by ramonjd** [comment]
> I believe I now wait for a Core Committer to merge this?

Yes! If @tellthemachines can't get to it today, hopefully someone will sweep in tonight and commit.

Thanks again!

**Comment by tellthemachines** [comment]
committed in [r56054](https://core.trac.wordpress.org/changeset/56054) / [14a3070](https://github.com/WordPress/wordpress-develop/commit/14a30703e550e8477a07d59879e55905dff3bd7c).

**Pull request closed without merging**
